modules/php/States/PlayCardTrait.php

```diff
@@ -228,6 +260,10 @@ public function playGoalCard($player_id, $card)
 
     // Notify all players about the goal played
     $goalCard = GoalCardFactory::getCard($card["id"], $card["type_arg"]);
+    
+    // this goal card is still in hand at this time
+    $handCount = $game->cards->countCardInLocation("hand", $player_id) - 1;
```

Comment:
Would it make sense to move this under the line 299 rather than doing a manual `-1` computation?

Reply:
Actually I thought here you had explicitly moved the notification up because otherwise the logs got confusing?
And we don't want to really play the goal already first, because that would make the Double Agenda check logic more complex and confusing.